### DISCUSSION TITLE
import word list

### DISCUSSION THREAD

**Discussion by paulobreim** [sepinf-inc/IPED#1779] [General]
This week I used the import word list function for a search.
This list had 39 words, and in the end iped-search reported that it read 39 words, but imported 38.
I checked one by one and realized that the word he didn't care about was because there was no occurrence of that word,. I was even surprised because the import speed is very fast, but beyond that it seems that it already checks for the existence of the word.
That's right?
So I wanted to suggest 2 things. It informs which was the word that didn't matter, or it imports everything even if there is no occurrence.

**Comment by lfcnassif** [comment]
> That's right?

Yes, this is the expected behavior.

**Reply by paulobreim** [reply]
@lfcnassif  At the end of import, it is possible to write a file with the words with no occurrences?

tks

**Reply by lfcnassif** [reply]
It is, but I think it would be better to show them in the dialog displayed after the import finishes.

**Reply by wladimirleite** [reply]
Writing words without occurrences to a file seems a very particular use case, and may have the inconvenience of requiring an action from the user (select where to save this file) or use a fixed location (which would be hidden). 

I think that showing a more detailed summary (like a small table with the number of occurrences per word) in the dialog displayed when importing finishes would be more flexible.
This dialog could have a button to copy to clipboard and possibly other options, like add items found to bookmarks (for each word).

**Reply by paulobreim** [reply]
Let me explain a situation that I have experienced more than once.

It is increasingly common, lawsuits where an employee leaves the company and goes to work at a competitor, and when leaving copies thousands of files that end up on the competitor's computer.

The victim files a search and seizure action, which he usually succeeds in, so that the existence of the subtracted files can be verified.
The defendant always tries to defend himself, claiming that the files cannot be seen since there are also files that are business secrets or private.

To alleviate the situation, I ask the company to provide a txt file with thousands of hashes referring to the files on the source computer.
So when examining the seized files, a simple search file load is enough to know whether or not there are identical files.
IPED doesn't just look up the hash, but there is a way to do that.
For example: in a scenario where the hash of the file on the seized computer is 2CE8A9605FD3C650129D6F385F1627A7
it is not found only with the hash, but if the imported file has a "hash:" in front of it, it will find it.
If in the txt I put:
2CE8A9605FD3C650129D6F385F1627A7
hash:2CE8A9605FD3C650129D6F385F1627A7
the first one is not imported, but the second one is located correctly.

Then with a simple import command, all the "stolen" files are quickly located without having to look further.

**Reply by lfcnassif** [reply]
Ok. So what is important to your use case is to know if the search criteria has hits, which the tool already does, and not if the search criteria doesn't have a hit, which you are asking for (not sure why), right?

**Reply by paulobreim** [reply]
Exact, it would just make documentation easier, but of course I can live without it.
